### PR TITLE
Issue 126 - Attempt to show 404 with existing template error tag

### PR DESCRIPTION
**Pull request by k10blogger**
As i had some extra time. I took it upon my self to do following changes with this PR.
1 - Implemented the exit first scenario wherever applicable and tried to bring everything as discuss in the coding guidelines. 
(Probably we should make a small wiki page with a coding guidelines)
2 - Showed how the error tag in the templates can be used for showing the 404 error. 

If we can decide upon a 404 error message we can provide support for multiple languages like `lang['404']` instead of a static text here:
image

Further improvements that are seen though not neccessary is that:
1 - There are 34 instances in the entire websvn where http_response_code is called with an error number (404|403|500)
We can take this opportunity to iumprove scenarios as:
image

2 - Basically this would mean extending the `renderTemplateNoRepo` to something more to add these error messages. Like additional arguments.

Let me know your thoughts.

### CODE REVIEW

**Comment by ams-tschoening** [comment]
One 404 error message only doesn't make sense, 404 is used for many different error conditions and each of those conditions need one error message. Each of those messages gets 404 status code, so `renderTemplateNoRepo` would become something like `renderTemplate404` with some `cause` or `errorDetail` or alike as argument. Sometimes that cause would be `NO_PATH`, sometimes it would be `NO_REPO` etc. The function would then do `$vars["error"] = $lang["..."]` and render one special error template most likely. That special error template would need to make sure that no controls are available for which the data is not available, e.g. because a repo is missing.

File: blame.php [hunk 1]
```diff
@@ -33,7 +33,7 @@
 // Make sure that we have a repository
 if (!$rep)
 {
-	renderTemplateNoRepo('blame');
+	renderTemplate404('blame',$lang['NOREP']);
```
Comment (ams-tschoening) [inline blame.php:36]: You should just forward the key `NOREP` and let `renderTemplate404` handle the lookup in `$lang`. This saves typing by the caller and at some point we might not associate an error message only with that key, like is the case currently, but e.g. a special template as well. You already made `$lang` available in `renderTemplate404` anyway, but most likely simply forgot to use it.

    renderTemplate404('blame', 'NOREP');

And:

    $vars['error'] = "WebSVN 404: " . $lang[$errorDetail];

**Review by ams-tschoening — CHANGES_REQUESTED** [review]

File: blame.php (hunk 1 above)
Comment (k10blogger) [reply blame.php:36]: ahh.. yes yes... this is my bad... I am fixing this...

Comment (k10blogger) [reply blame.php]: Done changed this.

**Review by ams-tschoening — APPROVED** [review]

**Pull request merged**